kafka/README.md

```diff
@@ -65,7 +67,38 @@ channels:
         topic: 'my-specific-topic-name'
         partitions: 20
         replicas: 3
-        bindingVersion: '0.3.0'
+        topicConfiguration:
+          cleanup.policy: ["delete", "compact"]
+          retention.ms: 604800000
+          retention.bytes: 1000000000
+          delete.retention.ms: 86400000
+          max.message.bytes: 1048588
+        bindingVersion: '0.4.0'
+```
+<a name="topicConfiguration"></a>
+## TopicConfiguration Object
+
+This objects contains information about the API relevant topic configuration in Kafka.
+
+Field Name |  Type   |                                                                             Description                                                                              | Applicability [default] | Constraints
+---|:-------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-----------------------:|---
+<a name="topicConfigurationCleanupPolicy"></a>`cleanup.policy` |  array  |                            The [`cleanup.policy`](https://kafka.apache.org/documentation/#topicconfigs_cleanup.policy) configuration option.                          |        OPTIONAL         | array may only contain `delete` and/or `compact`
+<a name="topicConfigurationRetentionMs"></a>`retention.ms` | integer |                               The [`retention.ms`](https://kafka.apache.org/documentation/#topicconfigs_retention.ms) configuration option.                                    |        OPTIONAL         | Must at least be -1
+<a name="topicConfigurationRetentionBytes"></a>`retention.bytes` | integer |                       The [`retention.bytes`](https://kafka.apache.org/documentation/#topicconfigs_retention.bytes) configuration option.                                                           |        OPTIONAL         | MUST at least be -1
+<a name="topicConfigurationDeleteRetentionBytes"></a>`delete.retention.ms` | integer |             The [`delete.retention.ms`](https://kafka.apache.org/documentation/#topicconfigs_delete.retention.ms) configuration option.                                               |        OPTIONAL         | MUST at least be 0
```

Comment:
as above, I think "at least" is possibly confusing. Not sure what the best way to phrase this, but essentially I'm trying to say that something like "If set, this MUST be 0 or a positive integer"